modules/utxo-ord/src/psbt.ts

```diff
@@ -135,19 +159,84 @@ export function findOutputLayoutForWalletUnspents(
       constraints.feeRateSatKB
     ),
   });
+
+  return layout ? { inputs, layout } : undefined;
 }
 
+/**
+ * @param inputs - inscription input must come first
+ * @param satPoint - location of the inscription
+ * @param outputs
+ * @param constraints
+ */
+function findSmallestOutputLayoutForWalletUnspents(
+  inputs: WalletUnspent[],
+  satPoint: SatPoint,
+  outputs: InscriptionTransactionOutputs,
+  constraints: InscriptionTransactionConstraints
+): { inputs: WalletUnspent[]; layout: OutputLayout } | undefined {
+  if (4 < inputs.length) {
+    throw new Error(`input array is too large`);
+  }
+  // create powerset of all supplementary inputs and find the cheapest result
+  const inputsArr = [inputs, ...powerset(inputs.slice(1)).map((s) => [inputs[0], ...s])];
+  return inputsArr
+    .map((inputs) => findOutputLayoutForWalletUnspents(inputs, satPoint, outputs, constraints))
+    .reduce((best, next) => {
+      if (best === undefined) {
+        return next;
+      }
+      if (next === undefined) {
+        return best;
+      }
+      return best.layout.feeOutput < next.layout.feeOutput ? best : next;
+    });
+}
+
+/**
+ * @param network
+ * @param inputBuilder
+ * @param unspent
+ * @param satPoint
+ * @param outputs
+ * @param constraints
+ * @param supplementaryUnspents - additional inputs to cover fee.
+ * @param [minimizeInputs=true] - try to find input combination with minimal fees. Limits supplementaryUnspents to 4.
+ */
 export function createPsbtForSingleInscriptionPassingTransaction(
   network: Network,
   inputBuilder: WalletInputBuilder,
-  unspents: bitgo.WalletUnspent<bigint>[],
+  unspent: WalletUnspent | WalletUnspent[],
   satPoint: SatPoint,
-  outputs: InscriptionOutputs,
-  constraints: InscriptionTransactionConstraints
+  outputs: InscriptionTransactionOutputs,
+  constraints: InscriptionTransactionConstraints,
+  {
+    supplementaryUnspents = [],
+    minimizeInputs = true,
+  }: {
+    supplementaryUnspents?: WalletUnspent[];
+    minimizeInputs?: boolean;
+  } = {}
 ): bitgo.UtxoPsbt {
-  const layout = findOutputLayoutForWalletUnspents(unspents, satPoint, outputs, constraints);
-  if (!layout) {
+  // support for legacy call style
+  if (Array.isArray(unspent)) {
+    if (unspent.length !== 1) {
```

Comment:
Can we ignore the latter unspents and just take the first one without throwing an error?